src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,109 @@
 /* eslint-disable */
-import React from 'react';
+import React, { useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { actions as currentTodoAction } from '../../features/currentTodo';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const todos = useAppSelector(state => state.todos);
+
+  const filteredTodos = () => {
+    const visibleTodos = [...todos];
+
+    return visibleTodos
+      .filter(todo =>
```

Comment:
> Also no need to filter in case query is empty
with such approach, you anyway filter todos, even if there is no query

Do something like that to avoid it

```
let filteredTodos = todos;

if (query.trim()) {
  filteredTodos = filteredTodos.filter(// filter by query)
}
```
Also, you may use similar approach to filter by status:
>Also no need to filter in case status all